esphome/components/mitsubishi_uart/__init__.py

```diff
@@ -149,9 +166,11 @@
             ),
             text_sensor.register_text_sensor,
         ),
-        "service_filter": (
-            "Service Filter",
-            binary_sensor.binary_sensor_schema(),
+        "filter_status": (
+            "Filter Status",
+            binary_sensor.binary_sensor_schema(
+                device_class="problem", icon="mdi:air-filter"
```

Comment:
Shouldn't this also be entity_category=ENTITY_CATEGORY_DIAGNOSTIC?

Reply:
So far nothing's been assigned to Diagnostic yet, I've been using it locally to track things like uptime and loop duration; the HA docs list RSSI and MAC address as examples of things that would go in Diagnostic.

I would say that values displayed on the thermostat to normal users are probably more in the "Sensors" category and more technical things like "Standby" and "Compressor Frequency" could be in "Diagnostic".  But then arguably "Standby" is something displayed directly to users on the wall units, so still probably more Sensor-y?

Might be worth further discussion for sensors as a whole.

Reply:
Yeah, the distinction is not super clear to me either, and I can see how "Filter status" is kind of borderline, unlike, say, Compressor Frequency. In my (limited) experience, HA & ESPHome components are all over the place in this matter.

If it helps, the distinction really is: diagnostic entities do not automatically show up in dashboards, are not exposed to Assist, and show up in a different section in the device page.